src/ui-kit/form-controls/date/date.component.ts

```diff
@@ -475,29 +475,26 @@ export class SamDateComponent
     let monthCheck = this.isMonthTouched || (!this.isMonthTouched && this.month.nativeElement.value);
     let yearCheck = this.isYearTouched || (!this.isYearTouched && this.year.nativeElement.value);
     if (dayCheck && monthCheck && yearCheck && !this.isTabPressed) {
-      if (this.month.nativeElement.value || this.day.nativeElement.value ||
-        this.year.nativeElement.value) {
-        if (this.isClean(override)) {
-          this.onChange(null);
-          this.valueChange.emit(null);
-        } else if (this.isYearTouched) {
-          if (this.year.nativeElement.value.length != 4) {
-            this.onChange('Invalid Date');
-            this.valueChange.emit('Invalid Date');
-          } else {
-            const dateString = this.getDate(override).format(this.OUTPUT_FORMAT);
-            this.onChange(dateString);
-            this.valueChange.emit(dateString);
-          }
-        } else if ((!this.getDate(override).isValid())) {
+      if (this.isClean(override)) {
```

Comment:
What about if it is dirty and you no longer want a date in the field (optional field)

Reply:
I'm not sure I understand this case. This check for isClean wasn't doing anything before, because `if (this.month.nativeElement.value || this.day.nativeElement.value || this.year.nativeElement.value)` guarantees that it would never be clean.

Reply:
What I am saying you want null as a response. Where is the case when you had a date in the field and than you delete all the fields?